servlet/src/main/java/io/grpc/servlet/ServletAdapter.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.servlet;
+
+import io.grpc.internal.GrpcUtil;
+import io.grpc.internal.ServerTransportListener;
+import java.io.IOException;
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * An adapter that transforms {@link HttpServletRequest} into gRPC request and lets a gRPC server
+ * process it, and transforms the gRPC response into {@link HttpServletResponse}. An adapter can be
+ * instantiated by {@link Factory#create}. The gRPC server is built from the ServerBuilder provided
+ * in {@link Factory#create}.
+ *
+ * <p>In a servlet, calling {@link #doPost(HttpServletRequest, HttpServletResponse)} inside {@link
+ * javax.servlet.http.HttpServlet#doPost(HttpServletRequest, HttpServletResponse)} makes the servlet
+ * backed by the gRPC server associated with the adapter. The servlet must support Asynchronous
+ * Processing and must be deployed to a container that supports servlet 4.0 and enables HTTP/2.
+ */
+public interface ServletAdapter {
```

Comment:
What's the point of this interface? Why don't we rename ServletAdaptorImpl to ServletAdaptor and use it instead?

Reply:
My thought was to be able to use CDI, i.e. `@Inject ServletAdapter servletAdapter`, in the servlet, and the servlet code needn't change when the producer is change from a grpc lib impl to a test object or to a container-vendor-provided impl. But it turned out that the `@PostConstruct` method to shutdown the server is not getting invoked if the producer uses `@Produces` method as in the example. So there is no advantage of using CDI for servletAdapter.

Changed the interface to a final class.